marshmallow/fields.py

```diff
@@ -405,22 +399,22 @@ def schema(self):
                     issubclass(self.nested, SchemaABC):
                 self.__schema = self.nested(
                     many=self.many,
-                    only=only, exclude=self.exclude, context=context,
+                    only=self.only, exclude=self.exclude, context=context,
                     load_only=self._nested_normalized_option('load_only'),
                     dump_only=self._nested_normalized_option('dump_only'))
             elif isinstance(self.nested, basestring):
                 if self.nested == 'self':
                     parent_class = self.parent.__class__
                     self.__schema = parent_class(
-                        many=self.many, only=only,
+                        many=self.many, only=self.only,
                         exclude=self.exclude, context=context,
                         load_only=self._nested_normalized_option('load_only'),
                         dump_only=self._nested_normalized_option('dump_only'))
                 else:
                     schema_class = class_registry.get_class(self.nested)
                     self.__schema = schema_class(
```

Comment:
Do you think the schema instantiation could be factorized to avoid repeating the schema kwargs? Or maybe just declare the kwargs dict only once?

Reply:
Ya, they are exactly the same.